common.sh

```diff
@@ -35,9 +35,29 @@ delete_pg()
     if [ ${ENABLE_PLACEMENT_GROUP} -eq 0 ] || [ -z $PLACEMENT_GROUP ]; then
         return 0
     fi
-    AWS_DEFAULT_REGION=us-west-2 aws ec2 delete-placement-group \
-        --group-name ${PLACEMENT_GROUP}
-    return $?
+    local ret=0
+    local retry=20
+    while [ "$retry" -ge 0 ]; do
+        AWS_DEFAULT_REGION=us-west-2 aws ec2 delete-placement-group \
+            --group-name ${PLACEMENT_GROUP}
+        ret=$?
+        # Error code 255 is InvalidPlacementGroup.InUse, which means the
+        # The placement group is in use and may not be deleted, due to
+        # the attached ec2 instances are not terminated completely.
+        if [ "$ret" -eq 255 ]; then
```

Comment:
This doesn't sound right.  Are you sure this is stable? 255 is -1, which makes me think other failures will also cause this.

Reply:
Thanks for catching it, I switched to check if the `InvalidPlacementGroup.InUse` is in the error message of this API call. 255 is definitely not limited for this error.